File: TLE Contest/Level 1 Contest 1/A_Playing_With_Palindrome.py
import math

n = int(input())
while n > 0:
    s = input()
    original = s
    if(len(s) <= 3):
        print("NO")
    else:
        temp = s[0]
        s[0] = s[1]
        s[1] = temp

        temp = s[-1]
        s[-1] = s[-2]
        s[-2] = temp

        if(original == s):
            print("NO")
        else:
            rev = s[::-1]
            if(s == rev):
                print("YES")
            else:
                print("NO")
    n -= 1

File: TLE Contest/Level 1 Contest 1/A_Playing_With_Palindrome.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;

    while (n--) {
        string s;
        cin >> s;
        string original = s;

        if (s.length() <= 3) {
            cout << "NO" << endl;
        }
        else {
            char temp = s[0];
            s[0] = s[1];
            s[1] = temp;

            temp = s[s.length() - 1];
            s[s.length() - 1] = s[s.length() - 2];
            s[s.length() - 2] = temp;

            if (original == s) {
                cout << "NO" << endl;
            }
            else {
                string rev = string(s.rbegin(), s.rend());
                if (s == rev) {
                    cout << "YES" << endl;
                }
                else {
                    cout << "NO" << endl;
                }
            }
        }
    }

    return 0;
}
